File: Frontend/src/app/service/archive.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Course } from "../model/course.model";
import { map } from 'rxjs/operators';
import { Archive } from "../model/archive.model";

@Injectable({
    providedIn:"root"
})
export class ArchiveService{

	private baseUrl = 'http://localhost:8080/archive';

    constructor(private httpClient:HttpClient){}

    getArchives():Observable<Archive[]>{
        return this.httpClient.get<Archive[]>(`${this.baseUrl}`);
    }

    getArchiveById(id:number):Observable<Archive>{
        return this.httpClient.get<Archive>(`${this.baseUrl}/${id}`);
    }

    addArchive(archive:Archive):Observable<any>{
        return this.httpClient.post<any>(`${this.baseUrl}`,archive, {responseType:'text' as 'json'}).pipe(map((resp)=>{
            return resp;
        }));
    }

    deleteArchive(id:number):Observable<any>{
        return this.httpClient.delete<any>(`${this.baseUrl}/${id}`,{responseType:'text' as 'json'}).pipe(map((resp)=>{
            return resp;
        }));
    }

    updateArchive(archive:Archive[]):Observable<any>{
        return this.httpClient.put<any>(`${this.baseUrl}`,archive,{responseType:'text' as 'json'}).pipe(map((resp)=>{
            return resp;
        }));
    }

    deduct(courseName:string, username:string):Observable<any>{
        return this.httpClient.delete<any>(`${this.baseUrl}/delete/${username}/${courseName}`,{responseType:'text' as 'json'}).pipe(map((resp)=>{
            return resp;
        }));
    }

    getArchiveByCourseName(courseName:string):Observable<Archive[]>{
        return this.httpClient.get<Archive[]>(`${this.baseUrl}/all/${courseName}`);
    }

    getArchiveByCourseNameAndUserName(courseName:string,username:string):Observable<Archive>{
        return this.httpClient.get<Archive>(`${this.baseUrl}/${courseName}/${username}`);
    }

}

File: Backend/src/main/java/by/grsu/course/service/ArchiveService.java
package by.grsu.course.service;

import by.grsu.course.model.Course;
import by.grsu.course.model.dto.ArchiveDTO;
import by.grsu.course.model.dto.CourseDTO;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.List;

public interface ArchiveService {
    List<ArchiveDTO> getAllArchive();
    ArchiveDTO getArchiveById(long id);
    void deleteArchive(long id);
    void updateArchive(List<ArchiveDTO> archiveDTO);
    void addNewArchive(ArchiveDTO archiveDTO);
    void deleteArchiveByCourseAndUserName(String courseName, String username);
    List<ArchiveDTO> getArchiveByCourseName(String courseName);
    ArchiveDTO getArchiveByCourseNameAndUserName(String courseName, String username);
}
